source/frontend/src/features/mapSideBar/management/tabs/activities/edit/InvoiceForm.tsx

```diff
@@ -0,0 +1,152 @@
+import { FormikProps } from 'formik';
+import React, { ChangeEvent, useEffect, useState } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import { FaTrash } from 'react-icons/fa';
+
+import { StyledRemoveLinkButton } from '@/components/common/buttons';
+import { FastCurrencyInput, FastDatePicker, Input, TextArea } from '@/components/common/form';
+import { YesNoSelect } from '@/components/common/form/YesNoSelect';
+import { Section } from '@/components/common/Section/Section';
+import { SectionField } from '@/components/common/Section/SectionField';
+
+import { ManagementActivityFormModel } from './models';
+
+export interface IInvoiceForm {
+  namespace: string;
+  index: number;
+  formikProps: FormikProps<ManagementActivityFormModel>;
+  onDelete: () => void;
+  gstConstant: number;
+  pstConstant: number;
+}
+
+export const InvoiceForm: React.FunctionComponent<IInvoiceForm> = ({
```

Comment:
Are there any differences between this form and the ad-hoc invoices form, besides the typing of the model?

Reply:
There are some difference in the main form (ManagementActivityForm) but the Invoice forms are the same. I was undecided whether to point to the ad-hoc ones or create a copy of them (which is what I did). I could go either way. Both invoice forms are the same but I tried to keep both "branches" separate. Otherwise the code is somewhat intertwined.

Again, I would be fine pointing to the other form but wanted to keep it separate in case they need to diverge later